plugins/modules/aci_l4l7_policy_based_redirect.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Tim Cragg (@timcragg)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: aci_l4l7_policy_based_redirect
+short_description: Manage L4-L7 Policy Based Redirection (vns:SvcRedirectPol)
+description:
+- Manage Layer 4 to Layer 7 (L4-L7) Policy Based Redirection
+options:
+  tenant:
+    description:
+    - The name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  policy_name:
+    description:
+    - The name of the Policy Based Redirection Policy.
+    type: str
+  description:
+    description:
+    - The description of the Policy Based Redirection Policy.
+    type: str
+  destination_type:
+    description:
+    - The destination type.
+    - The APIC defaults to C(l3) when unset during creation.
+    type: str
+    choices: [ l1, l2, l3 ]
+    aliases: [ dest_type ]
+  hash_algorithm:
+    description:
+    - The hashing algorithm.
+    - The APIC defaults to C(ip_and_protocol) when unset during creation.
+    type: str
+    choices: [ source_ip, destination_ip, ip_and_protocol ]
+  threshold_enable:
+    description:
+    - Whether to enable the threshold for the policy.
+    - The APIC defaults to C(false) when unset during creation.
+    type: bool
+  max_threshold:
+    description:
+    - The maximum percent when threshold is enabled.
+    - The APIC defaults to C(0) when unset during creation.
+    type: int
+  min_threshold:
+    description:
+    - The minimum percent when threshold is enabled.
+    - The APIC defaults to C(0) when unset during creation.
+    type: int
+  threshold_down_action:
+    description:
+    - The action to take when threshold is breached.
+    - The APIC defaults to C(permit) when unset during creation.
+    type: str
+    choices: [ deny, permit ]
+  resilient_hash:
+    description:
+    - Whether to enable resilient hashing.
+    - The APIC defaults to C(false) when unset during creation.
+    type: bool
+  pod_aware:
+    description:
+    - Whether to enable Pod ID aware redirection.
+    - The APIC defaults to C(false) when unset during creation.
+    type: bool
+  anycast_enabled:
+    description:
+    - Whether to enable anycast services.
+    - The APIC defaults to C(false) when unset during creation.
+    - Only available when I(destination_type=l3)
+    type: bool
+  monitor_policy:
```

Comment:
should we use full name?
```suggestion
  ip_sla_monitor_policy:
```